### PR TITLE
feat(linter): typescript/consistent-type-exports

### PR DESCRIPTION
**Pull request by EliLichtblau**
Sorry this includes my other PR, I totally borked this and honestly made a fork from a camp I was a part of in HS. 
But this adds types to the symbol tables, this will cause name conflicts so thats a todo, also I'm not really sure what I'm doing here, but I had to add types to implement at least a skeleton of consistent type imports. Which bans not annotating an export symbol as type when it is a type.

I.E: "type foo = number; export {foo}" is an error. 

This is mostly a wip and I'm largely looking for feedback before continuing.

### CODE REVIEW

**Comment by Boshen** [comment]
https://typescript-eslint.io/rules/consistent-type-exports/ says it requires type information.

Unfortunately we need to split this PR, it's hard for me to merge multiple things.

**Comment by EliLichtblau** [comment]
It doesn't really require type information, the information you need is symbol resolution. You need to trace the symbol and see if it resolves to a type (like trace all aliases and see if the declaration of the symbol is in a type-like err (non value like)). I sort of lazily implemented a type bindings for the symbol table, only looks at type alias declaration not interface because declaration merging. I'll see if it passes conformance.

File: crates/oxc_ast_lower/src/lib.rs [hunk 1]
```diff
@@ -41,7 +41,9 @@ impl<'a> AstLower<'a> {
         includes: SymbolFlags,
         excludes: SymbolFlags,
     ) -> SymbolId {
-        self.semantic.declare_symbol(span, name, includes, excludes)
+        self.semantic.declare_symbol(
+            span, name, includes, excludes, false, /* this is so much leaked complexity omg */
```
Comment (Boshen) [inline crates/oxc_ast_lower/src/lib.rs:45]: This comment is unnecessary, can you comment on `/* what this does */ false` instead?

**Comment by Boshen** [comment]
Can you rebase. There are also a lot of misspellings and informal comments which is not really nice for future readers.

This PR also adds type bindings to semantic, I need to take a closer look later.

**Pull request closed without merging**